packages/react-component-library/src/components/SelectE/SelectE.tsx

```diff
@@ -83,125 +68,114 @@ export const SelectE: React.FC<SelectEProps> = ({
-            aria-labelledby={labelId}
-          >
-            <StyledTextInput data-testid="text-input-container">
-              <StyledOuterWrapper
-                $hasFocus={isOpen}
-                $isDisabled={isDisabled}
-                $isInvalid={isInvalid}
-                data-testid="select-outer-wrapper"
-              >
-                <StyledInputWrapper data-testid="text-input-input-wrapper">
-                  <StyledLabel
-                    $hasContent={!!inputValue}
-                    $hasFocus={false}
-                    htmlFor={id}
-                    id={labelId}
-                    data-testid="select-label"
-                  >
-                    {label}
-                  </StyledLabel>
-                  <StyledInput
-                    $hasLabel
-                    data-testid="select-input"
-                    disabled={isDisabled}
-                    onMouseDown={() => {
-                      if (!isDisabled) {
-                        toggleMenu()
-                      }
-                    }}
-                    onFocus={() => {
-                      if (!isDisabled) {
-                        openMenu()
-                      }
-                    }}
-                    onMouseEnter={() => {
-                      if (!isDisabled) {
-                        setHasHover(true)
-                      }
-                    }}
-                    onMouseLeave={() => setHasHover(false)}
-                    readOnly
-                    {...rest}
-                    {...getInputProps()}
-                    id={id}
-                    aria-labelledby={labelId}
-                  />
-                </StyledInputWrapper>
-                <StyledInlineButtons>
-                  {inputValue && (
-                    <ClearButton
-                      isDisabled={isDisabled}
-                      onClick={() => clearSelection()}
-                    />
-                  )}
-                  <ArrowButton
-                    hasHover={hasHover}
-                    isDisabled={isDisabled}
-                    isOpen={isOpen}
-                    {...getToggleButtonProps()}
-                  />
-                </StyledInlineButtons>
-              </StyledOuterWrapper>
-            </StyledTextInput>
-            {isOpen && (
-              <StyledOptionsWrapper>
-                <StyledOptions {...getMenuProps()}>
-                  {React.Children.map(
-                    children,
-                    (child: SelectChildType, index: number) => {
-                      if (!React.isValidElement(child)) {
-                        return null
-                      }
+  const {
+    getItemProps,
+    getMenuProps,
+    getToggleButtonProps,
+    highlightedIndex,
+    isOpen,
+    openMenu,
+    selectItem,
+    selectedItem,
+    toggleMenu,
+  } = useSelect({
+    itemToString,
+    initialSelectedItem: React.Children.toArray(children).find((child) => {
```

Comment:
Is it worth breaking some of this stuff out into functions - this code is feels a little 'cramped'.